subsys/nrf_security/src/ssf_secdom/bounce_buffers.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/cache.h>
+
+#include "bounce_buffers.h"
+
+/* k_heap_alloc allocated memory is aligned on a multiple of pointer sizes. The HW's DataUnit size
+ * must match this Zephyr behaviour.
+ */
+BUILD_ASSERT(CACHE_DATA_UNIT_SIZE == sizeof(uintptr_t));
+
+static K_HEAP_DEFINE(out_buffer_heap,
+		     ROUND_UP(CONFIG_PSA_SSF_CRYPTO_CLIENT_OUT_HEAP_SIZE, CACHE_DATA_UNIT_SIZE));
+
+/**
+ * @brief Prepare an out buffer in case the original buffer is not aligned
+ *
+ * If the original buffer is not aligned, a new buffer is allocated and the data is copied to it.
+ * This is needed to achieve DCache DataUnit alignment.
+ *
+ * @param original_buffer Original buffer
+ * @param size Size of the buffer
+ * @return void* NULL if the buffer could not be allocated, original_buffer if it was aligned, else
+ * a new buffer from the heap
+ *
+ */
+void *bounce_buffers_prepare(void *original_buffer, size_t size)
+{
+	void *out_buffer;
```

Comment:
Shouldn't this be set explicitly to NULL?

Reply:
fixed